### PR TITLE
Desktop_Portfolio

### PR DESCRIPTION
**Pull request by AhmadiGu**
### Form local-storage

- add the local storage to store the form data.

### CODE REVIEW

File: form.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    emailError.innerHTML = '';
+    const data = JSON.stringify({ firstName,lastName, email, message });
+    localStorage.setItem('data', data);
+  }
+});
+const userData = JSON.parse(localStorage.getItem('data'));
+form.firstName.value = userData.name;
+form.lastName.value = userData.name;
+form.email.value = userData.email;
+form.message.value = userData.message;
```
Comment (Qoosim) [inline form.js]: - [REQUIRED] It is required to preserve the users data at the page load. Below features are what your form is expected to display.

-  After filling the form, and not submitting it, the entered values should be saved in the browser.
-  When visiting the page the form inputs should be contained, be auto-populated with, the values entered in the previous visit to the page.
-  If at the previous visit to the page, in the same browser, no input was entered in the form then the form inputs should remain empty.  
Kindly look into this and work your form to display these features.

At this moment, after filling the form with the appropriate data, if I reload the page without clicking on the submit button, I should still have my data preserved for me, but your form behaves contrary to that feature. Kindly make your form to display this feature.

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

-   No linter errors. :heavy_check_mark: 
-   Good README file. :heavy_check_mark: 
-   Correct GitHub Flow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

### Hint

You can use this as a guide

```
const formData = {
    name: nameInput.value,
    email: emailInput.value,
    body: bodyInput.value,
};


localStorage.setItem('contactFormInput', JSON.stringify(formData));

let savedFormData = localStorage.getItem('contactFormInput');
savedFormData = JSON.parse(savedFormData);

window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};

```

Best of luck :crossed_fingers:

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @AhmadiGu @obed300   ,
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Github Flow.
✔️ Professional README  
✔️ Good Commit Messages


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: form.js (hunk 1 above)
Comment (Abenezer-Tilahun) [reply form.js]: Fixed.

**Pull request merged**